data/src/main/java/co/nimblehq/compose/crypto/data/repository/CoinRepositoryImpl.kt

```diff
@@ -0,0 +1,35 @@
+package co.nimblehq.compose.crypto.data.repository
+
+import co.nimblehq.compose.crypto.data.model.response.toModels
+import co.nimblehq.compose.crypto.data.service.ApiService
+import co.nimblehq.compose.crypto.data.transform
+import co.nimblehq.compose.crypto.domain.model.CoinItem
+import co.nimblehq.compose.crypto.domain.repository.CoinRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.map
+
+class CoinRepositoryImpl(
+    private val api: ApiService
+) : CoinRepository {
+
+    override fun getTrendingList(
+        coinIds: List<String>,
+        currency: String,
+        priceChangePercentage: String,
+        itemOrder: String,
+        itemPerPage: Int,
+        page: Int
+    ): Flow<List<CoinItem>> = flow {
+        emit(
+            api.getCoins(
+                coinIds = coinIds.joinToString(separator = ","),
```

Comment:
Let's create a string in `UseCase` instead of putting it here. Also, Could you please create a `cont val` for separator`(,)` in UseCase as well 🙏 
```suggestion
                coinIds = coinIds,
```

Reply:
@Wadeewee For Api request, I think we should start handling the preparing request on Repository instead of UseCase. How do you think @luongvo ?

Reply:
> For Api request, I think we should start handling the preparing request on Repository instead of UseCase

@hoangnguyen92dn Hmm regarding this ☝️, If that, should we move this variable `myCoinIds` from `UseCase` to `Repository` instead 🤔 

But actually, We can create a request since `VM` and execute calling `API` via `UseCase.` So for the repository just get the request parameter that was filtered/created from `VM` via `UseCase` but currently, we don't integrate the VM yet, So It's good to put it in `UseCase` for now. Not sure you understand my request change, I mean we should call `joinToString(separator = ",")` since `UseCase` in line `33` 🙏

Reply:
@Wadeewee I got what you mean 😅 In my opinion, the method to combine the list of strings for request should be handled inside the Repository instead, because it requires for Api request not for any business purpose, for example the Backend changes the `Ids` field, and the separator is `-` (`btc-eth-...`), in this case, the update should not on the VM or UseCase, right? 😄

Reply:
I agree with @hoangnguyen92dn. The logic to map the id list into an id array string belongs to API request preparation, same to when preparing the data for request data class. So I think the current logic is fine 🤓 .